### DISCUSSION TITLE
Updating static page with redux state from localStorage (getStaticProps / getStaticPaths)

### DISCUSSION THREAD

**Discussion by sfhemstreet** [vercel/next.js#11847] [Help] [answered]
I am trying to build a fully static site with pages build using getStaticProps / getStaticPaths with content from a CMS.

I am using redux and saving its state in localStorage but I am running into an issue where pages with dynamic routes are not displaying the right UI, specifically while using styled-components. 

[This code sandbox demonstrates the problem I am running into.](https://codesandbox.io/s/funny-shannon-39j2m?file=/redux.js)

Follow these steps: 

1. Click a "product" and you will be directed to its page (products/[id]) made with getStaticProps / getStaticPaths.
2. Click the heart icon and the item will be added to the redux store and local storage.
3. Click "Go Back Home". You will see that the heart for the "product" is correctly filled in on the home page. 
4. Click the "product" again to go back to its page, the console will display an error "props don't match server" and the heart will not be filled in as it should. BUT the state will be correct.  When you click the heart it will remove the item.

This may be a problem with styled-components, but I am wondering if any of you have created a static site with redux and how you achieved it.

**Comment by ninjamovesonly** [comment, marked as answer]
You can't use localstorage on getServerSideProps or getStaticPaths as it runs in the server and does not have access to the window object like in the browser. 

I will suggest you use useEffect hook to get data from local storage
